docs/proposals/cri.md

```diff
@@ -0,0 +1,313 @@
+	conf.remoteRuntimeEndpoint = config.CONFIG.GetConfigurationByKey("edged.remote-runtime-endpoint").(string)
+	conf.remoteImageEndpoint = config.CONFIG.GetConfigurationByKey("edged.remote-image-endpoint").(string)
+	....
+}
+
+```
+### Runtime dependent module modifications
+The following modules which are dependent on the runtime needs to be handled during edged start based on docker runtime or
+remote CRI runtime.
+1. Volume Manager
+2. PLEG
+
+```go
+func (e *edged) Start(c *context.Context) {
+  ....
+  switch (e.containerRuntimeName) {
+    case DockerContainerRuntime:
+      e.volumeManager = volumemanager.NewVolumeManager(
+		    ....
+		    e.runtime.(*dockertools.DockerManager),
+        ....
+      )
+      e.pleg = edgedpleg.NewGenericLifecycle(
+        e.runtime.(*dockertools.DockerManager).ContainerManager,
+        .....
+      )
+    case RemoteContainerRuntime:
+      e.volumeManager = volumemanager.NewVolumeManager(
+		    ....
+		    e.containerRuntime,
+        ....
+      )
+      e.pleg = pleg.NewGenericLifecycle(
+        e.containerRuntime,
+        .....
+      )
+  }
+  ....
+    
+}
+```
+
+### Runtime dependent functional modifications
+
+The following functionalaties which are based on the docker runtime need to be modified to handle the CRI runtimes as well
+
+```go
+func (e *edged) initializeModules() error {
+  ....
+  switch (e.containerRuntimeName) {
+    case DockerContainerRuntime:
+	    e.runtime.Start(e.GetActivePods)
+    case RemoteContainerRuntime:
+      //TBD for CRI
+  ....
+}
+func (e *edged) consumePodAddition(namespacedName *types.NamespacedName) error {
+  ....
+  err = e.runtime.EnsureImageExists(pod, secrets)
+  //TBD for CRI
+  ....
+  err = e.runtime.StartPod(pod, opt)
+  // TBD for CRI
+  ....
+}
+
+func (e *edged) consumePodDeletion(namespacedName *types.NamespacedName) error {
+  ....
+  switch (e.containerRuntimeName) {
+    case DockerContainerRuntime:
+	    err := e.runtime.TerminatePod(pod.UID)
+    case RemoteContainerRuntime:
+      e.containerRuntime.KillPod(pod,nil,0)
+   }
+  
+  ....
+}
+
+func (e *edged) syncPod() {
+	//read containers from host
+	e.runtime.InitPodContainer()
+  
+  // TBD for CRI
+  
+  ....
+}
+
+func (e *edged) addPod(obj interface{}) {
+  ....
+  if (e.containerRuntimeName == DockerContainerRuntime)
+    e.runtime.UpdatePluginResources(nodeInfo, attrs)
+  ....
+}
+
+func (e *edged) HandlePodCleanups() error {
+  ....
+  switch (e.containerRuntimeName) {
+    case DockerContainerRuntime:
+	    containerRunningPods, err := e.runtime.GetPods(true)
+    case RemoteContainerRuntime:
+      containerRunningPods, err := e.containerRuntime.GetPods(true)
+  }
+  ....
+  if (e.containerRuntimeName == DockerContainerRuntime)
+    e.runtime.CleanupOrphanedPod(pods)
+  ....
+}
+
+```
+
+
+## Open questions
+- Do we need to support pause container and IP in edged ?
+- Do we need to support CNI in edged ?
```

Comment:
CNI support is out-of-scope in this document.

Reply:
Addressed